src/nflxmultisubs.js

```diff
@@ -14,14 +14,45 @@ const hookJsonParseAndAddCallback = function(_window) {
       if (result && result.result && result.result.movieId) {
           const movieId = result.result.movieId
           //console.log(`Intercepted manifest ${movieId}`);
-          window.__NflxMultiSubs.updateManifest(result.result);
+          // clone manifest so we can modify it w/o breaking player
+          let clonedManifest = _parse.call(JSON, ...args).result;
+          window.__NflxMultiSubs.updateManifest(clonedManifest);
+          if (gRenderOptions.secondaryLanguageMode === 'last') {
+            if (result.result.timedtexttracks.find(t=>t.rank==0).language === gRenderOptions.secondaryLanguageLastUsed) {
+              result.result.recommendedMedia.timedTextTrackId = result.result.timedtexttracks.find(t=>t.rank==1).new_track_id;
+              console.log("Restored recommended primary subtitle");
+            }
+          }
       }
       return result;
   };
 };
 hookJsonParseAndAddCallback(window);
 
 
+// Hook JSON.stringify() to intercept and modify manifest requests
+const hookJsonStringify = function(_window) {
+  const _stringify = JSON.stringify;
+  _window.JSON.stringify = (...args) => {
+    if (args[0] && (args[0].url === 'licensedManifest' || args[0].url === 'manifest')) {
+      // key name is different for different cadium-playercore versions (e.g. 'Fb' for 6.0033.414.911, 'Db' for 6.0038.181.911, 'zb' for 6.0034.588.911)
+      // so we check each one by testing two common properties
+      for (let i in args[0]) {
+        if (args[0][i] && args[0][i].viewableId && args[0][i].manifestVersion) {
+          if (gRenderOptions.secondaryLanguageMode === 'last') {
+            args[0][i].preferredTextLocale =  gRenderOptions.secondaryLanguageLastUsed;
+            console.log('Modified request to get last sub language', gRenderOptions.secondaryLanguageLastUsed);
+          }
```

Comment:
Same issue as above. Setting `preferredTextLocale` is working great and it is correctly requesting the last used language so that it shows up in the secondary language list even if it is an 'extra' language. But: the side effect is that it will also set that language as the primary recommended language and activate the subtitles for it. If you can't immediately find a workaround for it I would prefer to leave it out for now.